### PR TITLE
[EX-108] Add generic types instead of any

### PR DESCRIPTION
**Pull request by skyman503**

### CODE REVIEW

File: example/src/types/MetadataTypes.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+export type TrackMetadataType = {
```
Comment (graszka22) [inline example/src/types/MetadataTypes.ts:1]: do we need to add `Type` suffix here? isn't it obvious that a type is a type?

File: src/index.tsx [hunk 2]
```diff
@@ -196,18 +196,18 @@ export function useWebRTC() {
  * This hook provides live updates of room endpoints.
  * @returns An array of room endpoints.
  */
-export function useEndpoints() {
-  const [endpoints, setEndpoints] = useState<Endpoint[]>([]);
+export function useEndpoints<T extends Metadata, S extends Metadata>() {
```
Comment (graszka22) [inline src/index.tsx:199]: can't T and S be more descriptive? I don't know what do they mean

Comment (graszka22) [reply src/index.tsx:199]: also probably the metadata can be different for different tracks

for example, an audio track can have 'muted' property, which doesn't make sense for camera track

File: src/index.tsx [hunk 3]
```diff
@@ -508,22 +510,26 @@ export function useScreencast() {
...
+export async function updateEndpointMetadata<T extends Metadata>(metadata: T) {
   await MembraneWebRTCModule.updateEndpointMetadata(metadata);
 }
 
 /**
  * a function that updates video metadata on the server.
  * @param metadata a map string -> any containing video track metadata to be sent to the server
  */
-export async function updateVideoTrackMetadata(metadata: Metadata) {
+export async function updateVideoTrackMetadata<T extends Metadata>(
```
Comment (graszka22) [inline src/index.tsx:521]: there is a problem with that: I can specify different type for metadata when starting the camera and when updating camera track metadata
ideally it'd be inferred automatically but I don't know how to do this or if it's possible (maybe with some api changes?)

**Review by graszka22 — COMMENTED** [review]
I added some comments, treat them rather like proposals, some of them maybe doesn't have sense ;) 
in general let's also think if adding those types improved user experience or it's just an additional hassle

File: example/src/types/MetadataTypes.ts (hunk 1 above)
Comment (skyman503) [reply example/src/types/MetadataTypes.ts]: Removed.

**Review by graszka22 — APPROVED** [review]
LGTM 👍 
Also let Kamil take a look

**Comment by skyman503** [comment]
@kamil-stasiak  PTAL

**Review by kamil-stasiak — APPROVED** [review]

**Pull request merged**